src/App.tsx

```diff
@@ -1,41 +1,76 @@
-import React from 'react';
+import { FC, useEffect, useState } from 'react';
 import './App.scss';
 import './styles/general.scss';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
+import { Post } from './types/Post';
+import { getUserPosts } from './api/posts';
+import { Loader } from './components/Loader';
+import { apiHelper } from './api/apiHelper';
 
-const App: React.FC = () => (
-  <div className="App">
-    <header className="App__header">
-      <label>
-        Select a user: &nbsp;
+const App: FC = () => {
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [userId, setUserId] = useState(0);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMsg, setErrorMsg] = useState('');
+  const [selectedPostId, setSelectedPostId] = useState(0);
 
-        <select className="App__user-selector">
-          <option value="0">All users</option>
-          <option value="1">Leanne Graham</option>
-          <option value="2">Ervin Howell</option>
-          <option value="3">Clementine Bauch</option>
-          <option value="4">Patricia Lebsack</option>
-          <option value="5">Chelsey Dietrich</option>
-          <option value="6">Mrs. Dennis Schulist</option>
-          <option value="7">Kurtis Weissnat</option>
-          <option value="8">Nicholas Runolfsdottir V</option>
-          <option value="9">Glenna Reichert</option>
-          <option value="10">Leanne Graham</option>
-        </select>
-      </label>
-    </header>
+  useEffect(() => {
+    apiHelper(
+      getUserPosts,
+      userId,
+      setIsLoading,
+      setErrorMsg,
+    ).then(setPosts);
+  }, [userId]);
 
-    <main className="App__main">
-      <div className="App__sidebar">
-        <PostsList />
-      </div>
+  return (
+    <div className="App">
+      <header className="App__header">
+        <label>
+          Select a user: &nbsp;
 
-      <div className="App__content">
-        <PostDetails />
-      </div>
-    </main>
-  </div>
-);
+          <select
+            className="App__user-selector"
+            value={userId}
+            onChange={(event) => setUserId(Number(event.target.value))}
+          >
+            <option value="0">All users</option>
+            <option value="1">Leanne Graham</option>
+            <option value="2">Ervin Howell</option>
+            <option value="3">Clementine Bauch</option>
+            <option value="4">Patricia Lebsack</option>
+            <option value="5">Chelsey Dietrich</option>
+            <option value="6">Mrs. Dennis Schulist</option>
+            <option value="7">Kurtis Weissnat</option>
+            <option value="8">Nicholas Runolfsdottir V</option>
+            <option value="9">Glenna Reichert</option>
+            <option value="10">Leanne Graham</option>
```

Comment:
let's, mb, try to fetch users?
try to query all the posts, and get users from them 🤔

Reply:
I thought about it🙂

Reply:
> let's, mb, try to fetch users? try to query all the posts, and get users from them 🤔

for what? there are a lot of users with an empty post list

Reply:
>for what? there are a lot of users with an empty post list

so what? 😄 
it is still better to have an array of users, then hardcode them 😉